subt/main.py

```diff
@@ -201,15 +207,19 @@ def go_safely(self, desired_direction):
         else:
             self.send_speed_cmd(desired_speed, desired_angular_speed)
 
-    def turn(self, angle, with_stop=True, speed=0.0):
+    def turn(self, angle, with_stop=True, speed=0.0, timeout=None):
         print(self.time, "turn %.1f" % math.degrees(angle))
         start_pose = self.last_position
         if angle >= 0:
             self.send_speed_cmd(speed, self.max_angular_speed)
         else:
             self.send_speed_cmd(speed, -self.max_angular_speed)
+        start_time = self.time
         while abs(normalizeAnglePIPI(start_pose[2] - self.last_position[2])) < abs(angle):
             self.update()
+            if timeout is not None and self.time - start_time > timeout:
+                print("turn - TIMEOUT!")
```

Comment:
Can you add `self.time` to the beginning of the line?

Reply:
fixed.